os.cpp

```diff
@@ -37,6 +39,23 @@ void dump(bool dumpRegs)
 
 }
 
+void printQueue(queue<User> &s,int num)
+{
+    if(!num)
+    {
+        cout << endl;
+        return;
+    }
+    User x= s.front();
+    s.pop();
+    cout << "\nUSER " << x.id << " REGISTERS";
+		cout << "\n\trA: " << x.regs.rA << ", r1: " << x.regs.r1 << ", r2: " \
+			<< x.regs.r2 << ", r3: " << x.regs.r3 << endl;
+		cout << "\tIR: " << x.regs.IR << ", PC: " << x.regs.PC << ", CR: " << x.regs.CR << endl;
+    s.push(x);
+    printQueue(s,--num);
```

Comment:
I like the recursion with specified size. I'd have never thought of that :D 

I'm going to edit the function so it takes a parameter `title` as a string, and then function can then output it's own little header. That way you'll know which queue is being printed, and which users belong where, and it will all be encapsulated in the printQueue function, rather than having random cout << [header] statements everywhere.